connectors/src/connectors/google_drive/temporal/activities.ts

```diff
@@ -555,117 +581,94 @@ export async function renewWebhooks(pageSize: number): Promise<number> {
-    where: {
-      expiresAt: {
-        [Op.lt]: literal("now() - INTERVAL '1 day'"),
-      },
-      renewedByWebhookId: {
-        [Op.not]: null,
-      },
-    },
-    limit: pageSize,
-  });
-
-  return webhooks.length;
-}
-
-export async function renewOneWebhook(webhookId: ModelId) {
-  const wh = await GoogleDriveWebhook.findByPk(webhookId);
-  if (!wh) {
-    throw new Error(`Webhook ${webhookId} not found`);
-  }
-
-  const connector = await ConnectorResource.fetchById(wh.connectorId);
-
-  if (connector) {
-    try {
-      const webhookInfo = await registerWebhook(connector);
-      if (webhookInfo.isErr()) {
-        throw webhookInfo.error;
-      } else {
-        await sequelizeConnection.transaction(async (t) => {
-          const freshWebhook = await GoogleDriveWebhook.create(
-            {
-              webhookId: webhookInfo.value.id,
-              expiresAt: new Date(webhookInfo.value.expirationTsMs),
-              renewAt: new Date(webhookInfo.value.expirationTsMs),
-              connectorId: connector.id,
-            },
-            { transaction: t }
-          );
-          await wh.update(
+    const connector = await ConnectorResource.fetchById(wh.connectorId);
+    if (connector) {
+      try {
+        const res = await ensureWebhookForDriveId(
+          connector,
+          wh.driveId,
+          GOOGLE_DRIVE_WEBHOOK_RENEW_MARGIN_MS
+        );
+        if (res.isErr()) {
+          // Throwing here to centralize error handling in the catch block
+          // below.
+          throw res.error;
+        }
+        if (res.value) {
+          // marking the previous webhook as renewed by the new one.
+          await wh.update({
+            renewedByWebhookId: res.value,
+          });
+        }
+      } catch (e) {
+        if (
+          e instanceof ExternalOauthTokenError ||
+          (e instanceof HTTPError &&
+            e.message === "The caller does not have permission")
+        ) {
+          await syncFailed(connector.id, "oauth_token_revoked");
+          logger.error(
             {
-              renewedByWebhookId: freshWebhook.webhookId,
+              error: e,
+              connectorId: wh.connectorId,
+              workspaceId: connector.workspaceId,
             },
-            {
-              transaction: t,
-            }
+            `Failed to renew webhook: Oauth token revoked .`
           );
-        });
-      }
-    } catch (e) {
-      if (
-        e instanceof ExternalOauthTokenError ||
-        (e instanceof HTTPError &&
-          e.message === "The caller does not have permission")
-      ) {
-        await syncFailed(connector.id, "oauth_token_revoked");
+          // Do not delete the webhook object but push it down the line in 2h so that it does not get
+          // picked up by the loop calling rewnewOneWebhook.
+          await wh.update({
+            renewAt: literal("NOW() + INTERVAL '2 hour'"),
+          });
+          continue;
```

Comment:
Extra nit: I think an if/else would be more readable and less error prone than a if/continue